saleor/graphql/attribute/utils.py

```diff
@@ -144,19 +144,22 @@ def _pre_save_values(
         cls, attribute: attribute_models.Attribute, attr_values: AttrValuesInput
     ):
         """Lazy-retrieve or create the database objects from the supplied raw values."""
-        get_or_create = attribute.values.get_or_create
-
+        result = []
         if not attr_values.values:
             return tuple()
-
-        return tuple(
-            get_or_create(
-                attribute=attribute,
-                slug=slugify(value, allow_unicode=True),
-                defaults={"name": value},
-            )[0]
-            for value in attr_values.values
-        )
+        for value in attr_values.values:
+            value_obj = attribute.values.filter(name=value).first()
```

Comment:
Here we run a DB query in each iteration. Can we optimize this? E.g. build some dict with all the needed data before entering the for-loop, with one DB query?

Reply:
If there would be more values with matching slugs, the filtering and creating unique slug would not work correctly... Or am I wrong? @IKarbowiak

Reply:
Yep. If somebody will provide `red` and `RED` we will get the same slug from `generate_unique_slug` and we will get the error when saving. We can optimize that but in such cases, we need to handle that provided values may end up with the same slug